### DISCUSSION TITLE
gql string with embedded expression

### DISCUSSION THREAD

**Discussion by earnubs** [graphql-hive/graphql-eslint#579] [Q&A] [answered]
When linting a template literal with an embedded expression, graphql-eslint suffers a parse failure because the field is empty at the point of parsing:

```
gql`
    enum Foo {
        ${big_list_of_things_I_computed_elsewhere}
    }
`
```

Is this something I can address via config or a feature request?

**Comment by ardatan** [comment]
Related https://github.com/ardatan/graphql-tools/issues/1732

@dotansimha @B2o5T Maybe we should ignore this kind of codes automatically. There is no way to parse these.

@earnubs You can add ignore annotations to that code for now because we will never be able to parse it correctly.

**Reply by dimaMachina** [reply]
All right! I think we need improve that. Maybe test with regular expression if there is string interpolation in the documents and ignore them?

**Reply by earnubs** [reply]
@ardatan I don't think you can ignore these because it's a parse failure? but maybe I'm doing it wrong, I'm assuming you mean something like wrapping the enum in ` # eslint-disable / enable` comments?

**Reply by earnubs** [reply]
@B2o5T With a regex to skip, I think the next problem I'd have is from the point of view of eslint, the enum/type would be empty in this case. So it might mean that the empty type rule would also need to be relaxed at this point too?

**Reply by dimaMachina** [reply]
I think eslint comment `# eslint-disable` don’t work because `graphql-tag-pluck` receive code like
```gql
enum Foo {

}
```
And this is invalid code because enum don’t contain any fields, so parser breaks. I think if you add path to this file to `.eslintignore` eslint will completely ignore this file for parsing and linting

**Reply by earnubs** [reply]
What I meant was:

```
# eslint-disable
enum Foo {
  ${stuff}
}
# eslint-enable
```

but this doesn't work, I think, because it would need to be parsed first in order to work, and we have a parse error :)

Ignoring the whole file does work, it's never parsed in that case.

**Reply by earnubs** [reply]
The other things that does "work", but is not useful to me (but maybe someone else), is adding an obviously wrong value to make eslint happy:

```
enum Foo {
 __IGNORE_ME__
 ${stuff}
}

**Comment by dimaMachina** [comment, marked as answer]
Hi @earnubs, currently it’s not possible because for extracting documents from code under the hood we use [graphql-tag-pluck](https://graphql-tools.com/docs/graphql-tag-pluck) that [don't support string interpolation](https://stackoverflow.com/questions/62749847/graphql-codegen-dynamic-fields-with-interpolation/62751311#62751311) for this moment.
